src/app/products/manage-tax-groups/create-tax-group/create-tax-group.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators, FormArray } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ProductsService } from 'app/products/products.service';
+import { MatDialog } from '@angular/material';
+import { CancelDialogComponent } from 'app/shared/cancel-dialog/cancel-dialog.component';
+import { DatePipe } from '@angular/common';
+
+@Component({
+  selector: 'mifosx-create-tax-group',
+  templateUrl: './create-tax-group.component.html',
+  styleUrls: ['./create-tax-group.component.scss']
+})
+export class CreateTaxGroupComponent implements OnInit {
+  TaxGroupForm: FormGroup;
+    /** tax Component Data. */
+    taxComponentData: any;
+
+  constructor(private formBuilder: FormBuilder,
+              private route: ActivatedRoute,
+              private router: Router,
+              private datePipe: DatePipe,
+              public dialog: MatDialog,
+              private productService: ProductsService) {}
+
+  ngOnInit() {
+    // Create initial form
+    this.createTaxGroupForm();
+    this.productService.getTaxComponents().subscribe((data: any) => {
+      this.taxComponentData = data;
+    });
+  }
+
+  // Creates tax group form
+  createTaxGroupForm() {
+    this.TaxGroupForm = this.formBuilder.group({
+      'name': ['', Validators.required],
+      'taxComponents': this.formBuilder.array([])
+    });
+  }
+
+  /**
+   * Gets the taxcomponent form array.
+   * @returns {FormArray} taxcomponent form array.
+   */
+  get taxComponents(): FormArray {
+    return this.TaxGroupForm.get('taxComponents') as FormArray;
+  }
+
+
+  createTaxcomponentForm(): FormGroup {
+    return this.formBuilder.group({
+      'taxComponentId': [''],
+      'startDate': ['']
+    });
+  }
+
+  addTaxcomponent(): void {
+    this.taxComponents.push(this.createTaxcomponentForm());
+  }
+
+  /**
+   * Cancels the group form creation and redirects to Tax Group Managing page.
+   */
+  cancelTaxGroupForm() {
+    const cancelTaxGroupDialogRef = this.dialog.open(CancelDialogComponent);
+    cancelTaxGroupDialogRef.afterClosed().subscribe((response: any) => {
+      if (response.cancel) {
+        this.router.navigate(['../'], {relativeTo: this.route});
+      }
+    });
+  }
+
+  /**
+   * Submits the survey form and creates survey,
+   * if successful redirects to surveys.
+   */
+  submit() {
+    const prevstartdate: Date = this.TaxGroupForm.value.startDate;
```

Comment:
```suggestion
    const prevStartdate: Date = this.TaxGroupForm.value.startDate;
```
Change other instances accordingly.